watchtower/wtclient/client.go

```diff
@@ -0,0 +1,722 @@
+		// fires.
+		//
+		// For full safety, this can be set to 0 and wait out
+		// indefinitely.  However for mobile clients which may have a
+		// limited amount of time to exit before the background process
+		// is killed, this offers a way to ensure the process
+		// terminates.
+		if c.cfg.ForceQuitDelay > 0 {
+			time.AfterFunc(c.cfg.ForceQuitDelay, c.ForceQuit)
+		}
+
+		// 3. Once the backup queue has shutdown, wait for the main
+		// dispatcher to exit. The backup queue will signal it's
+		// completion to the dispatcher, which releases the wait group
+		// after all tasks have been assigned to session queues.
+		c.wg.Wait()
+
+		// 4. Since all valid tasks have been assigned to session
+		// queues, we no longer need to negotiate sessions.
+		c.negotiator.Stop()
+
+		log.Debugf("Waiting for active session queues to finish "+
+			"draining, stats: %s", c.stats)
+
+		// 5. Shutdown all active session queues in parallel. These will
+		// exit once all updates have been acked by the watchtower.
+		c.activeSessions.ApplyAndWait(func(s *sessionQueue) func() {
+			return s.Stop
+		})
+
+		// Skip log if force quitting.
+		select {
+		case <-c.forceQuit:
+			return
+		default:
+		}
+
+		log.Debugf("Client successfully stopped, stats: %s", c.stats)
+	})
+	return nil
+}
+
+// ForceQuit idempotently initiates an unclean shutdown of the watchtower
+// client. This should only be executed if Stop is unable to exit cleanly.
+func (c *TowerClient) ForceQuit() {
+	c.forced.Do(func() {
+		log.Infof("Force quitting watchtower client")
+
+		// Cancel log message from stop.
+		close(c.forceQuit)
+
+		// 1. Shutdown the backup queue, which will prevent any further
+		// updates from being accepted. In practice, the links should be
+		// shutdown before the client has been stopped, so all updates
+		// would have been added prior.
+		c.pipeline.ForceQuit()
+
+		// 2. Once the backup queue has shutdown, wait for the main
+		// dispatcher to exit. The backup queue will signal it's
+		// completion to the dispatcher, which releases the wait group
+		// after all tasks have been assigned to session queues.
+		c.wg.Wait()
+
+		// 3. Since all valid tasks have been assigned to session
+		// queues, we no longer need to negotiate sessions.
+		c.negotiator.Stop()
+
+		// 4. Force quit all active session queues in parallel. These
+		// will exit once all updates have been acked by the watchtower.
+		c.activeSessions.ApplyAndWait(func(s *sessionQueue) func() {
+			return s.ForceQuit
+		})
+
+		log.Infof("Watchtower client unclean shutdown complete, "+
+			"stats: %s", c.stats)
+	})
+}
+
+// RegisterChannel persistently initializes any channel-dependent parameters
+// within the client. This should be called during link startup to ensure that
+// the client is able to support the link during operation.
+func (c *TowerClient) RegisterChannel(chanID lnwire.ChannelID) error {
+	// TODO(conner): create sweep pkscript for new channels
```

Comment:
Mean to only make a new pkScript for that channel?

Reply:
yes, so whenever register is called the first time we will derive and persist a new pkscript to use for that channel. if one has already been made, this will be a nop

Reply:
ended up adding this to the current working PR